STY: Remove trailing whitespace

MartinThoma opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
* https://github.com/sympy/sympy/issues/17287

#### Brief description of what is fixed or changed

All trailing whitespaces were removed from the codebase

#### Other comments

It was done with the following `.pre-commit-config.yaml`:

```
repos:
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v2.3.0
    hooks:
    -   id: trailing-whitespace
        exclude: ^sympy/parsing/latex/_antlr/|^sympy/parsing/autolev/_antlr/|^sympy/integrals/rubi/
```

by running `pre-commit run --all-files`. No other change was done.


#### Release Notes

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

oscarbenjamin commented:
I'm not sure if some of the autolev/antlr code is auto-generated. It's probably best to ignore any files that are already ignored by flake8/mypy (see setup.cfg). Not sure if you already have ignored them...

MartinThoma committed:
STY: Remove trailing whitespace

MartinThoma commented:
Good point, @oscarbenjamin . I originally didn't and I reverted the changes to them and adjusted the example pre-commit in the post above. Now only files which are not ignored by flake8 (via `setup.cfg`) should have been touched.

asmeurer commented on doc/man/isympy.1:
```diff
@@ -169,14 +169,14 @@ from the other isympy options.
 Print isympy version information and exit.
 .SH FILES
-.TP 
+.TP
```
Is trailing whitespace significant in troff files?

MartinThoma replied:
Good point. I assume(d) that it would not have an effect, but it's maybe better to ask: https://stackoverflow.com/q/61487473/562769 (and check; but I don't know how)

asmeurer commented:
This looks fine, other than the question about the manpage. Maybe we should extend the trailing whitespace test to check all relevant files in the repo (assuming the test is running under git). I think the main point of avoiding trailing whitespace is to avoid unnecessary churn in diffs, which applies equally to Python and non-Python files.

MartinThoma commented:
> This looks fine, other than the question about the manpage

What is the question about the manpage?

smichr commented:
> This looks fine, other than the question about the manpage.

looks like response on SO affirms that spaces removed here are not an issue; if there is no objection I think this can be merged

smichr commented:
@oscarbenjamin , this could be included in the 1.6 release

oscarbenjamin commented:
Yes, see https://github.com/sympy/sympy/pull/19236#issuecomment-623158406

iammosespaulr commented:
#19249 
I created a PR from this branch to 1.6, and there are no release notes so, should be fine.


> @oscarbenjamin , this could be included in the 1.6 release

MartinThoma commented:
I'm a bit confused about this... can it be merged? Is there anything left to be done?

MartinThoma commented:
Cool, thank you :-)

sylee957 commented:
After removing the trailing whitespace in manpage, it always shows the diff after `make man` is run.

MartinThoma commented:
Ah, I didn't know that the manpage is automatically generated. I guess then it should be undone.
